samza-core/src/main/java/org/apache/samza/execution/StreamPlan.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.execution;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import org.apache.samza.config.JobConfig;
+import org.apache.samza.operators.StreamGraph;
+import org.apache.samza.system.StreamSpec;
+
+
+/**
+ * This class represents Samza {@link org.apache.samza.application.StreamApplication}
+ * plans for physical execution.
+ */
+public class StreamPlan {
+  private final StreamGraph streamGraph;
+  private final JobGraph jobGraph;
```

Comment:
Commentary (no action in this PR):
This is the most compelling evidence yet that StreamGraph needs to be renamed. When I first read this, I thought the StreamPlan was keeping separate graphs to track the streams and the jobs. 

Now if they were named more meaningfully, like 
ApplicationLogic
JobGraph

I'd have a much better idea of what this plan was tracking.